test/parallel/test-tls-inception.js

```diff
@@ -28,6 +28,10 @@ a = tls.createServer(options, function (socket) {
   var dest = net.connect(options);
   dest.pipe(socket);
   socket.pipe(dest);
+
+  dest.on('close', function() {
+    socket.destroy();
+  });
```

Comment:
Ah! that's the problem. `dest` is a streams3 stream, which does *not* call `destroy` on downstream streams on `close`. streams1 streams *would* call destroy on downstream streams.